### ISSUE TITLE
grepping typo in scanProcess function

### ISSUE THREAD

**Issue by aircrack-ng** [aircrack-ng/aircrack-ng#1733]
**Reported by slava phirsov on 7 Nov 2017 19:27 UTC**

Look at lines 1075..1076 of the script airmon-ng:

```
#openwrt busybox grep hits on itself so we -v it out
match="$(ps ${psopts} | grep -c ${PROCESSES} | grep -v grep)"
```

The ```grep -v grep``` filter is needed to exclude grep itself from counted processes list. But wait... This filter input is ```grep -c ${PROCESSES} ``` output, containing match count. So, the first (counting) filter counts itself, and the second filter filters nothing. Solution? Simple: just twist two filters

```
match="$(ps ${psopts} | grep -v grep | grep -c ${PROCESSES})"
```

**Comment by aircrack-ng** [comment]
**Comment by misterx on 7 Nov 2017 19:27 UTC**

You're right, it is incorrect but the solution is not good, it will still count the grep process.

Here is a better one:

```
match="$(ps ${psopts} | grep ${PROCESSES} | wc -l)"
```

What do you think?

**Comment by aircrack-ng** [comment]
**Comment by slava phirsov on 7 Nov 2017 19:27 UTC**

Thinking about trying both snippets on Busybox, but have no ability to do it right now. Your solution counts the number of matching lines in ps output using intended wc utility. But the grep itself is not excluded from the ps output and count. So, merging both solutions:

```
match="$(ps ${psopts} | grep -v grep | grep ${PROCESSES} | wc -l)"
```

**Comment by aircrack-ng** [comment]
**Comment by slava phirsov on 7 Nov 2017 19:27 UTC**

Apropos: does openwrt busybox support wc utility?

**Comment by aircrack-ng** [comment]
**Comment by slava phirsov on 7 Nov 2017 19:27 UTC**

AFAIK ```grep ${PATTERN} | wc -l``` does the same things as ```grep -c ${PATTERN} ```, so, I think, wc usage is redundant.

**Comment by aircrack-ng** [comment]
**Comment by anonymous on 7 Nov 2017 19:27 UTC**

My bad. wc is supported but I would still swap both grep:

```
match="$(ps ${psopts} | grep ${PROCESSES} | grep -v grep | wc -l)"
```

**Comment by aircrack-ng** [comment]
**Comment by misterx on 7 Nov 2017 19:27 UTC**

Fixed in aircrack-ng/aircrack-ng@7552fdc8181cc74fc487077cee40eb672ba696eb.
